client/lib/happychat/test/index.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { stub } from 'sinon';
+import { EventEmitter } from 'events';
+
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_CONNECTING,
+	HAPPYCHAT_CONNECTED,
+	HAPPYCHAT_DISCONNECTED,
+	HAPPYCHAT_RECEIVE_EVENT,
+	HAPPYCHAT_RECONNECTING,
+	HAPPYCHAT_SET_AVAILABLE,
+	HAPPYCHAT_SET_CHAT_STATUS,
+	HAPPYCHAT_TRANSCRIPT_REQUEST,
+} from 'state/action-types';
+
+import buildConnection from '../connection';
+
+describe( 'connection', ( ) => {
+	describe( 'should bind socket ', ( ) => {
+		const user = {
+			signer_user_id: 12,
+			jwt: 'jwt',
+			locale: 'locale',
+			groups: 'groups',
+			geo_location: 'location',
+		};
+
+		let openSocket, socket, dispatch;
+		beforeEach( () => {
+			socket = new EventEmitter();
+			dispatch = stub();
+			const connection = buildConnection();
+			openSocket = connection.init( socket, dispatch, user );
+		} );
+
+		it( 'connect event', ( done ) => {
+			openSocket.then( ( ) => {
+				// TODO: implement when connect event is used
+				expect( true ).to.equal( true );
+				done(); // tell mocha the promise chain ended
+			} );
+			socket.emit( 'connect' );
+			socket.emit( 'init' ); // force openSocket promise to resolve
+		} );
+
+		it( 'token event', ( done ) => {
+			const callback = stub();
+			openSocket.then( ( ) => {
+				expect( callback.getCall( 0 ).args[ 0 ].signer_user_id ).to.equal( user.signer_user_id );
```

Comment:
We should probably store `callback.getCall( 0 ).args[ 0 ]` in a variable to cut down on the repetition

Reply:
you could actually make this a little easier with something like

```js
const { signer_user_id, jwt, locale, groups } = callback.getCall( 0 ).args[ 0 ];
expect( user ).to.include( { signer_user_id, jwt, locale, groups } );
```

Reply:
`calledWithMatch` is nice! Fixed all tests in 4646789